docs/development/development.md

```diff
@@ -0,0 +1,76 @@
+</details>
+
+---
+
+### **How do the rules for immutable fields/migrations work and where are they evaluated in furyctl's code?**
+
+<details>
+  <summary>Answer</summary>
+
+The rules for immutable fields and migrations are evaluated within the core logic of the `furyctl` tool. These rules are defined in the configuration files and enforced by the tool to ensure consistency and prevent misconfigurations.
+
+These rules serve as safety mechanisms during module changes (e.g., switching from the Loki logging system to OpenSearch). Some changes are allowed, while others are not. For example, fields marked as `immutable` will return an error if an attempt is made to change them.
+
+The rules are configured in the `rules` folder, and the commands/scripts executed for each rule are found in `templates/distribution/scripts/pre-apply.sh.tpl`.
+
+Additionally, **Reducers** are special fields rendered inside the template engine that indicate whether a particular feature or module of the distribution has changed. The `.to` and `.from` strings indicate these changes precisely.
+
+</details>
+
+---
+
+### **What's the dependency evaluation flow in the `kfd.yaml` and how does furyctl download them?**
+
+<details>
+  <summary>Answer</summary>
+
+The `kfd.yaml` file, specific to the distribution downloaded by `furyctl`, contains the definitions of dependencies that the distribution relies on.
+
+Furyctl reads the version information embedded within the `kfd.yaml` file to determine which dependencies need to be fetched. These dependencies are typically other resources or libraries required for the successful deployment or operation of the current distribution.
+
+Once identified, `furyctl` downloads or references these dependencies from either a local or remote repository. This ensures that the distribution is fully equipped with all the necessary components for deployment, preventing version mismatches and compatibility issues. The process ensures the environment is set up with the correct versions and configurations.
+
+</details>
+
+---
+
+### **What's inside the Makefile?**
+
+<details>
+  <summary>Answer</summary>
+
+The important commands in the `Makefile` are:
+
+- **`make generate-go-models`**: This command generates Go code from the JSON schema files. The generated code defines the data models used in the codebase, providing a structured representation of the resources and configurations used by `furyctl`. It essentially converts the schema into Go structs, which are essential for interacting with the configuration data programmatically.
+
+- **`make generate-docs`**: This command generates Markdown documentation from the schema files. It extracts the necessary information from the schemas and formats it into human-readable documentation, helping developers and users understand how to configure and use the distribution and resources. This documentation serves as the primary reference for anyone interacting with `furyctl`.
```

Comment:
let's add some infos on how to jumpstart the environment to be able to run these two commands